### PR TITLE
ci: fix removed syntax for GHA

### PR DESCRIPTION
**Pull request by henryiii**
Actions replaced this due to a security venerability: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

Should fix CI issue in #27 

Closes #22.

### CODE REVIEW

**Comment by henryiii** [comment]
Still broken because of https://github.com/excitedleigh/setup-nox/issues/235 .

**Comment by henryiii** [comment]
Well, pre-commit is now running. :) I managed the GHA fix from my phone, but not about to run pre-commit here. ;)

**Comment by pradyunsg** [comment]
Yea, this project probably needs some TLC.

**Comment by henryiii** [comment]
@pradyunsg Passes now. Only code change was the one in ed62023 to make MyPy happy. 
b307347 was updating pre-commit (it was on MyPy 7.400! Basically couldn't read the current, valid type annotations.) And the pre-commit run output is in the final commit.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,33 +1,36 @@
...
+      - id: check-toml
+      - id: check-yaml
+      - id: debug-statements
+      - id: end-of-file-fixer
+      - id: forbid-new-submodules
+      - id: trailing-whitespace
 
   - repo: https://github.com/psf/black
-    rev: stable
+    rev: 20.8b1
```
Comment (henryiii) [inline .pre-commit-config.yaml:16]: Never point pre-commit at a moving tag/branch! It was designed to into its caching system. Instead, run pre-commit autoupdate or turn on pre-commit.ci, which updates weekly as needed via PR.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,33 +1,36 @@
 repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
```
Comment (henryiii) [inline .pre-commit-config.yaml:2]: Always put checks that modify above checks that do not (flake8, mypy)

File: src/vendoring/utils.py [hunk 3]
```diff
@@ -30,11 +30,12 @@ def run(command: List[str], *, working_directory: Optional[Path]) -> None:
...
-        line = p.stdout.readline().rstrip()
+        if p.stdout is not None:
+            line = p.stdout.readline().rstrip()
 
-        if line:
-            with UI.indent():
-                UI.log(line)
+            if line:
+                with UI.indent():
+                    UI.log(line)
```
Comment (henryiii) [inline src/vendoring/utils.py:38]: This is the one change, to make MyPy happy. The poll can return None in theory (thought it won't in this case). Not sure why this was a poll in the first place?

**Comment by henryiii** [comment]
Happy to rebase #27 after this one.

**Comment by pradyunsg** [comment]
I ended up revamping the entire setup yesterday! Thanks for the nudge and reminder that the CI was completely borked here. :)

**Pull request closed without merging**